src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQuerySQLEngine.java

```diff
@@ -218,18 +224,24 @@ public SQLDataset join(SQLJoinRequest sqlJoinRequest) throws SQLEngineException
                                                                       dataset,
                                                                       runId);
 
+    LOG.info("Executed join operation for dataset {}", sqlJoinRequest.getDatasetName());
+
     datasets.put(sqlJoinRequest.getDatasetName(), joinDataset);
 
     return joinDataset;
   }
 
   @Override
   public void cleanup(String datasetName) throws SQLEngineException {
+
     BigQuerySQLDataset bqDataset = datasets.get(datasetName);
+    //if (bqDataset == null || config.skipDelete()) {
```

Comment:
unintended change?

Reply:
Yes! This is something I plan to introduce later as a way to add a few debugging behaviors to the plugin.